internal/olm/operator/registry/fbcindex/fbc_registry_pod.go

```diff
@@ -254,13 +261,44 @@ func (f *FBCRegistryPod) podForBundleRegistry(cs *v1alpha1.CatalogSource) (*core
 			},
 		})
 
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{
-			Name:      k8sutil.TrimDNS1123Label(cm.Name + "-volume"),
+		if f.cfg.GzipCM {
+			volumes = append(volumes, corev1.Volume{
+				Name: k8sutil.TrimDNS1123Label(cm.Name + "-unzip"),
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			})
+		}
+
+		postfix := cm.Name + "-volume"
+		if f.cfg.GzipCM {
+			postfix = "-unzip"
+		}
+
+		vm := corev1.VolumeMount{
+			Name:      k8sutil.TrimDNS1123Label(cm.Name + postfix),
 			MountPath: path.Join(f.FBCIndexRootDir, cm.Name),
 			SubPath:   cm.Name,
-		})
+		}
+
+		if !f.cfg.GzipCM {
+			vm.SubPath = cm.Name
+		}
+
+		sharedVolumeMounts = append(sharedVolumeMounts, vm)
+
+		if f.cfg.GzipCM {
+			gzipVolumeMount = append(gzipVolumeMount, corev1.VolumeMount{
+				Name:      k8sutil.TrimDNS1123Label(cm.Name + "-volume"),
+				MountPath: path.Join("/compressed", f.FBCIndexRootDir, cm.Name),
+				SubPath:   cm.Name,
+			})
+		}
 	}
 
+	containerVolumeMount := make([]corev1.VolumeMount, len(sharedVolumeMounts))
+	copy(containerVolumeMount, sharedVolumeMounts)
```

Comment:
Why do we need to do this copy? Can we just pass sharedVolumeMounts directly to wherever we are passing containerVolumeMount?

Reply:
We don't. it's a leftover from a previous implementation. Removed.